### ISSUE TITLE
Question about Training

### ISSUE THREAD

**Issue by ChaoGaoUCR** [TUDB-Labs/multi-lora-fine-tune#91]
Dear Authors,

Thanks for this great project.
I got a question about training, 
I can see this part only produces one work for inference, why are we not using auto-regressive here?
Also, I wondered how we should test the throughput like tokens per second.

Best
Chao
<img width="827" alt="Screenshot 2023-11-01 at 2 01 50 PM" src="https://github.com/TUDB-Labs/multi-lora-fine-tune/assets/86901408/a4331736-1023-4421-a15e-c0881d0e4d7e">

**Comment by yezhem** [comment]
* The [repo](https://github.com/yezhem/aspen-experiment) show how to test the throughput.
* This project is now just used for the train, the inference's implementation will be added later(the main branch's inference just for test).

**Comment by ChaoGaoUCR** [comment]
Thanks for replying to me.
Yes, I understand the patch.
I just got confused by the feed-forward part in the tuning session,
aren't we supposed to do auto-regression in the feed-forward parts?

**Comment by yezhem** [comment]
Do you mean why there is only use the output to train, not like the inference produces token by token and calc the MSELoss?

**Comment by ChaoGaoUCR** [comment]
Yes,
I was thinking aren't we supposed to first generate a series of outputs and then use the output tokens to compare with the ground truth label?
Like: "ML is Fun" as input, Out put is "ML is Fun topic to work on" which add 4 words(maybe 12 tokens)
Then the ground truth is "ML is Fun topic indeed"
Then we compare these two.

**Comment by yezhem** [comment]
I think sequence classification/regression does this, if we just train for chat, cross-entropy loss maybe better. 
If your input is "ML is Fun topic indeed.", the output will contain all predict, "ML xx", "ML is xx", "ML is Fun xx"
But we can add the classification/regression features later if needed.

**Comment by merlintang** [comment]
since no comments, i would like to close this one.
